src/ApacheOrcDotNet.OptimizedReader/Buffers/BaseColumnBuffer.cs

```diff
@@ -0,0 +1,285 @@
+﻿using ApacheOrcDotNet.Encodings;
+using ApacheOrcDotNet.OptimizedReader.Encodings;
+using ApacheOrcDotNet.OptimizedReader.Infrastructure;
+using ApacheOrcDotNet.Protocol;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace ApacheOrcDotNet.OptimizedReader.Buffers
+{
+    [SkipLocalsInit]
+    public abstract class BaseColumnBuffer<TOutput>
+    {
+        private readonly long[] _numericStreamBuffer;
+        private readonly byte[] _byteStreamBuffer;
+        private readonly byte[] _boolStreamBuffer;
+
+        private protected readonly IByteRangeProvider _byteRangeProvider;
+        private protected readonly OrcContext _context;
+        private protected readonly OrcColumn _column;
+        private protected readonly TOutput[] _values;
+
+        private protected readonly ArrayPool<byte> _pool;
+
+        private protected int _numValuesRead;
+
+        private StreamRange _lastRange;
+        private int _lastRangeLength;
+
+        public BaseColumnBuffer(IByteRangeProvider byteRangeProvider, OrcContext context, OrcColumn column)
+        {
+            _byteRangeProvider = byteRangeProvider;
+            _context = context;
+            _column = column;
+            _values = new TOutput[_context.MaxValuesToRead];
+
+            _pool = ArrayPool<byte>.Create(15 * 1024 * 1024, 8);
```

Comment:
The first is the `maxArrayLength` and the second one is the `maxArraysPerBucket`. They would match the default values we had for compression/decompression buffers.

P.S.: We will `new` up the buffers that are currently being rented, so this will be removed.